pkg/apis/apps/validation/validation.go

```diff
@@ -96,8 +98,21 @@ func ValidatePetSet(petSet *apps.PetSet) field.ErrorList {
 // ValidatePetSetUpdate tests if required fields in the PetSet are set.
 func ValidatePetSetUpdate(petSet, oldPetSet *apps.PetSet) field.ErrorList {
 	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&petSet.ObjectMeta, &oldPetSet.ObjectMeta, field.NewPath("metadata"))...)
-	allErrs = append(allErrs, ValidatePetSetSpec(&petSet.Spec, field.NewPath("spec"))...)
+
+	// TODO: For now we're taking the safe route and disallowing all updates to spec except for Spec.Replicas.
+	// Enable on a case by case basis.
+	restoreReplicas := petSet.Spec.Replicas
+	petSet.Spec.Replicas = oldPetSet.Spec.Replicas
+
+	// The generation changes for this update
+	restoreGeneration := petSet.Generation
+	petSet.Generation = oldPetSet.Generation
+
+	if !reflect.DeepEqual(petSet, oldPetSet) {
```

Comment:
yeah but we use it all over the place in our validation (https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L1383). It's leaves less room for error than a hash.